tools/bug-report/pkg/content/content.go

```diff
@@ -0,0 +1,219 @@
+	}
+)
+
+// Params contains parameters for running a kubectl fetch command.
+type Params struct {
+	Client    kube.ExtendedClient
+	DryRun    bool
+	Verbose   bool
+	Namespace string
+	Pod       string
+	Container string
+}
+
+func (p *Params) SetClient(client kube.ExtendedClient) *Params {
+	out := *p
+	out.Client = client
+	return &out
+}
+
+func (p *Params) SetDryRun(dryRun bool) *Params {
+	out := *p
+	out.DryRun = dryRun
+	return &out
+}
+
+func (p *Params) SetVerbose(verbose bool) *Params {
+	out := *p
+	out.Verbose = verbose
+	return &out
+}
+
+func (p *Params) SetNamespace(namespace string) *Params {
+	out := *p
+	out.Namespace = namespace
+	return &out
+}
+
+func (p *Params) SetPod(pod string) *Params {
+	out := *p
+	out.Pod = pod
+	return &out
+}
+
+func (p *Params) SetContainer(container string) *Params {
+	out := *p
+	out.Container = container
+	return &out
+}
+
+func retMap(filename, text string, err error) (map[string]string, error) {
+	if err != nil {
+		return nil, err
+	}
+	return map[string]string{
+		filename: text,
+	}, nil
+
+}
+
+// GetK8sResources returns all k8s cluster resources.
+func GetK8sResources(p *Params) (map[string]string, error) {
+	out, err := kubectlcmd.RunCmd("get --all-namespaces "+
+		"all,jobs,ingresses,endpoints,customresourcedefinitions,configmaps,events "+
+		"-o yaml", "", p.DryRun)
+	return retMap("k8s-resources", out, err)
+}
+
+// GetSecrets returns all k8s secrets. If full is set, the secret contents are also returned.
+func GetSecrets(p *Params) (map[string]string, error) {
+	cmdStr := "get secrets --all-namespaces"
+	if p.Verbose {
+		cmdStr += " -o yaml"
+	}
+	out, err := kubectlcmd.RunCmd(cmdStr, "", p.DryRun)
+	return retMap("secrets", out, err)
+}
+
+// GetCRs returns CR contents for all CRDs in the cluster.
+func GetCRs(p *Params) (map[string]string, error) {
+	crds, err := getCRDList(p)
+	if err != nil {
+		return nil, err
+	}
+	out, err := kubectlcmd.RunCmd("get --all-namespaces "+strings.Join(crds, ",")+" -o yaml", "", p.DryRun)
+	return retMap("crs", out, err)
+}
+
+// GetClusterInfo returns the cluster info.
+func GetClusterInfo(p *Params) (map[string]string, error) {
+	out, err := kubectlcmd.RunCmd("cluster-info dump", "", p.DryRun)
+	return retMap("cluster-info", out, err)
+}
+
+// GetDescribePods returns describe pods for istioNamespace.
+func GetDescribePods(p *Params) (map[string]string, error) {
+	if p.Namespace == "" {
+		return nil, fmt.Errorf("getDescribePods requires the Istio namespace")
+	}
+	out, err := kubectlcmd.RunCmd("describe pods", p.Namespace, p.DryRun)
+	return retMap("describe-pods", out, err)
+}
+
+// GetEvents returns events for all namespaces.
+func GetEvents(params *Params) (map[string]string, error) {
+	out, err := kubectlcmd.RunCmd("get events --all-namespaces -o wide", "", params.DryRun)
+	return retMap("events", out, err)
+}
+
+// GetIstiodInfo returns internal Istiod debug info.
+func GetIstiodInfo(p *Params) (map[string]string, error) {
```

Comment:
can use `clientAllDiscoveryDo`? the implementation is more robust/tested/efficient

Reply:
and lower privs